### ISSUE TITLE
New and Existing Signs not displaying after 18 hours on new spigot 1.8.

### ISSUE THREAD

**Issue by mactso** [webbukkit/dynmap#1718]
We went to minecraft spigot 1.8.1 sunday night.  Signs displayed.
On Monday, after about 18 hours the signs stopped displaying.
I tested this on the test server and (same spigot, same dynmap) the signs display on the map.
This may be a glitch.  I'm considering reinstalling dynmap and adding the markers back in manually if it is working.

Three sections Follow
Description of Problem
Log File
Config File

Description of Problem
When we create a [dynmap] sign in the game, the [dynmap] still disappears so it looks like they are being processed.  New signs do not display either.  Also the icons are not displaying.

I've done a partial and a fullrender to see if that would address the problem.
We've restarted the server and the signs do not display.  The map is otherwise working correctly (chat, updates, appearance).

Log File
There are no errors.
The only warning is 
[12:57:58] [Server thread/WARN]: Plugin `RailMob_2.0 v2.0' uses the space-character (0x20) in its name `RailMob 2.0' - this is discouraged

Here's every [dynmap] message from the server log.

[12:57:58] [Server thread/INFO]: [dynmap] Loading dynmap v2.1-SNAPSHOT-1077
[12:57:58] [Server thread/INFO]: [dynmap] version=git-Spigot-59b08be-0899683 (MC: 1.8)
[12:57:58] [Server thread/INFO]: [dynmap] inhabitedTicks field not found - inhabited shader not functional
[12:57:58] [Server thread/INFO]: [dynmap] Mod Support API available
[12:58:04] [Server thread/INFO]: [dynmap] Enabling dynmap v2.1-SNAPSHOT-1077
[12:58:04] [Server thread/INFO]: [dynmap] Using Bukkit Permissions (superperms) for access control
[12:58:04] [Server thread/INFO]: [dynmap] Web interface permissions only available for online users
[12:58:04] [Server thread/INFO]: [dynmap] Mod Support processing completed
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 23 shaders.
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 82 perspectives.
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 13 lightings.
[12:58:05] [Server thread/INFO]: [dynmap] Web server started on address 0.0.0.0:8123
[12:58:05] [Server thread/INFO]: [dynmap] version 2.1-SNAPSHOT-1077 is enabled - core version 2.1-SNAPSHOT-2021
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 1 maps of world 'HardWorld'.
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 5 pending tile renders for world 'HardWorld'
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 1 maps of world 'HardWorld_nether'.
[12:58:05] [Server thread/INFO]: [dynmap] Loaded 2 maps of world 'HardWorld_the_end'.
[12:58:05] [Server thread/INFO]: [dynmap] Enabled
[12:58:09] [Dynmap Render Thread/INFO]: [dynmap] Loading default resource pack

Here is the dynmap Config File
http://pastebin.com/txEXwSds

**Comment by mactso** [comment]
Oh yes, people's head's and chat balloons are displaying.

**Comment by mactso** [comment]
This is the marker file.
icons: {
    }
sets:
    markers:
        circles: {
            }
        hide: false
        markers:
            marker_1:
                icon: temple
                markup: false
                label: <Center>
                z: 31.0
                world: HardWorld_nether
                y: 65.0
                x: 11.0
            marker_2:
                icon: temple
                markup: false
                label: <CommunityCenter>
                z: 249.0
                world: HardWorld
                y: 64.0
                x: 126.0
            _sign_HardWorld_596_65_82:
                icon: sign
                markup: true
                label: Dryad Fountain<br/>Blessing
                z: 82.5
                world: HardWorld
                y: 65.5
                x: 596.5
            _sign_HardWorld_-535_75_198:
                icon: sign
                markup: true
                label: Hill Gate
                z: 198.5
                world: HardWorld
                y: 75.5
                x: -534.5
            _sign_HardWorld_761_69_-4969:
                icon: sign
                markup: true
                label: Farpoint
                z: -4968.5
                world: HardWorld
                y: 69.5
                x: 761.5
            _sign_HardWorld_2042_64_-6361:
                icon: sign
                markup: true
                label: Northern<br/>Mushroom<br/>Island
                z: -6360.5
                world: HardWorld
                y: 64.5
                x: 2042.5
            _sign_HardWorld_nether_23_65_135:
                icon: sign
                markup: true
                label: Portal to:<br/>Southern Horse<br/>Path
                z: 135.5
                world: HardWorld_nether
                y: 65.5
                x: 23.5
            _sign_HardWorld_41_63_382:
                icon: sign
                markup: true
                label: Super-mega<br/>Wheat Farm!
                z: 382.5
                world: HardWorld
                y: 63.5
                x: 41.5
            _sign_HardWorld_33_150_159:
                icon: sign
                markup: true
                label: Opulence <br/>Meeting Area
                z: 159.5
                world: HardWorld
                y: 150.5
                x: 33.5
            _sign_HardWorld_29_152_257:
                icon: sign
                markup: true
                label: Castle Opulence<br/>I Has It
                z: 257.5
                world: HardWorld
                y: 152.5
                x: 29.5
            _sign_HardWorld_nether_-3_35_34:
                icon: sign
                markup: true
                label: Portal to:<br/>PWiddy's Zombie<br/>Spawner
                z: 34.5
                world: HardWorld_nether
                y: 35.5
                x: -2.5
            _sign_HardWorld_nether_17_65_36:
                icon: sign
                markup: true
                label: Portal to:<br/>Community<br/>Center
                z: 36.5
                world: HardWorld_nether
                y: 65.5
                x: 17.5
            _sign_HardWorld_nether_95_61_-326:
                icon: sign
                markup: true
                label: Plymouth
                z: -325.5
                world: HardWorld_nether
                y: 61.5
                x: 95.5
            _sign_HardWorld_184_69_339:
                icon: sign
                markup: true
                label: Punky Brewster
                z: 339.5
                world: HardWorld
                y: 69.5
                x: 184.5
            _sign_HardWorld_1511_67_-2981:
                icon: sign
                markup: true
                label: Goofy Cave
                z: -2980.5
                world: HardWorld
                y: 67.5
                x: 1511.5
            _sign_HardWorld_-814_87_-890:
                icon: sign
                markup: true
                label: Hesiod II
                z: -889.5
                world: HardWorld
                y: 87.5
                x: -813.5
            _sign_HardWorld_755_60_-3525:
                icon: sign
                markup: true
                label: Awesome Mine
                z: -3524.5
                world: HardWorld
                y: 60.5
                x: 755.5
            _sign_HardWorld_184_64_1743:
                icon: sign
                markup: true
                label: Mesa Biome<br/>Homestead
                z: 1743.5
                world: HardWorld
                y: 64.5
                x: 184.5
            _sign_HardWorld_134_70_414:
                icon: sign
                markup: true
                label: Falleopia's<br/>Farm
                z: 414.5
                world: HardWorld
                y: 70.5
                x: 134.5
            _sign_HardWorld_nether_-51_18_30:
                icon: sign
                markup: true
                label: Portal to:<br/>MactsoCo<br/>from Stronghold
                z: 30.5
                world: HardWorld_nether
                y: 18.5
                x: -50.5
            _sign_HardWorld_57_151_191:
                icon: sign
                markup: true
                label: Opulence Garden
                z: 191.5
                world: HardWorld
                y: 151.5
                x: 57.5
            _sign_HardWorld_nether_124_37_1133:
                icon: sign
                markup: true
                label: Portal to:<br/>Southern<br/>Ice Spikes
                z: 1133.5
                world: HardWorld_nether
                y: 37.5
                x: 124.5
            _sign_HardWorld_2140_64_-7561:
                icon: sign
                markup: true
                label: Redcap<br/>Mushroom<br/>Big Cave
                z: -7560.5
                world: HardWorld
                y: 64.5
                x: 2140.5
            _sign_HardWorld_-2062_122_228:
                icon: sign
                markup: true
                label: Lost Wither<br/>Sailing Beach
                z: 228.5
                world: HardWorld
                y: 122.5
                x: -2061.5
            _sign_HardWorld_nether_-120_75_24:
                icon: sign
                markup: true
                label: Portal to:<br/>Port Cog
                z: 24.5
                world: HardWorld_nether
                y: 75.5
                x: -119.5
            _sign_HardWorld_38_76_315:
                icon: sign
                markup: true
                label: Bargain Berna
                z: 315.5
                world: HardWorld
                y: 76.5
                x: 38.5
            _sign_HardWorld_2126_64_-7726:
                icon: sign
                markup: true
                label: underwater <br/>big cave
                z: -7725.5
                world: HardWorld
                y: 64.5
                x: 2126.5
            _sign_HardWorld_-403_67_257:
                icon: sign
                markup: true
                label: MactsoCo<br/>Trading Center
                z: 257.5
                world: HardWorld
                y: 67.5
                x: -402.5
            _sign_HardWorld_122_64_252:
                icon: sign
                markup: true
                label: The Center<br/>Community<br/>Center
                z: 252.5
                world: HardWorld
                y: 64.5
                x: 122.5
            _sign_HardWorld_-2826_69_-2140:
                icon: sign
                markup: true
                label: Desert Temple<br/>Gate
                z: -2139.5
                world: HardWorld
                y: 69.5
                x: -2825.5
            _sign_HardWorld_nether_263_34_-906:
                icon: sign
                markup: true
                label: Mushroom Cave
                z: -905.5
                world: HardWorld_nether
                y: 34.5
                x: 263.5
            _sign_HardWorld_34_74_300:
                icon: sign
                markup: true
                label: Vizard's Tower<br/>(Quests)
                z: 300.5
                world: HardWorld
                y: 74.5
                x: 34.5
            _sign_HardWorld_4396_69_-118:
                icon: sign
                markup: true
                label: Eastern<br/>Ice Spikes
                z: -117.5
                world: HardWorld
                y: 69.5
                x: 4396.5
            _sign_HardWorld_nether_-153_68_169:
                icon: sign
                markup: true
                label: Blaze Spawner<br/>&amp;<br/>Nether Fortress
                z: 169.5
                world: HardWorld_nether
                y: 68.5
                x: -152.5
            _sign_HardWorld_188_66_290:
                icon: sign
                markup: true
                label: The Hot House
                z: 290.5
                world: HardWorld
                y: 66.5
                x: 188.5
            _sign_HardWorld_nether_-354_76_-274:
                icon: sign
                markup: true
                label: Desert Temple<br/>Gate
                z: -273.5
                world: HardWorld_nether
                y: 76.5
                x: -353.5
            _sign_HardWorld_748_72_-2604:
                icon: sign
                markup: true
                label: test 1
                z: -2603.5
                world: HardWorld
                y: 72.5
                x: 748.5
            _sign_HardWorld_12_34_276:
                icon: sign
                markup: true
                label: PWiddy's<br/>Zombie Spawner<br/>Mk. II!
                z: 276.5
                world: HardWorld
                y: 34.5
                x: 12.5
            _sign_HardWorld_-616_71_435:
                icon: sign
                markup: true
                label: Mt. MactsoCo
                z: 435.5
                world: HardWorld
                y: 71.5
                x: -615.5
            _sign_HardWorld_nether_-77_39_-901:
                icon: sign
                markup: true
                label: Ghastwood
                z: -900.5
                world: HardWorld_nether
                y: 39.5
                x: -76.5
            _sign_HardWorld_-1109_67_1128:
                icon: sign
                markup: true
                label: Forest Outpost
                z: 1128.5
                world: HardWorld
                y: 67.5
                x: -1108.5
            _sign_HardWorld_9621_56_2103:
                icon: sign
                markup: true
                label: Tupt Sea
                z: 2103.5
                world: HardWorld
                y: 56.5
                x: 9621.5
            _sign_HardWorld_-584_70_-5514:
                icon: sign
                markup: true
                label: SandCave
                z: -5513.5
                world: HardWorld
                y: 70.5
                x: -583.5
            _sign_HardWorld_nether_-137_76_144:
                icon: sign
                markup: true
                label: Portal to:<br/>Forest Outpost
                z: 144.5
                world: HardWorld_nether
                y: 76.5
                x: -136.5
            _sign_HardWorld_742_79_-3273:
                icon: sign
                markup: true
                label: future<br/>speed boost
                z: -3272.5
                world: HardWorld
                y: 79.5
                x: 742.5
            _sign_HardWorld_nether_548_71_-15:
                icon: sign
                markup: true
                label: Portal to:<br/>Eastern<br/>Ice Spikes
                z: -14.5
                world: HardWorld_nether
                y: 71.5
                x: 548.5
            _sign_HardWorld_59_86_278:
                icon: sign
                markup: true
                label: The Ender God<br/>Temple
                z: 278.5
                world: HardWorld
                y: 86.5
                x: 59.5
            _sign_HardWorld_-212_65_-35:
                icon: sign
                markup: true
                label: Casa C7ron
                z: -34.5
                world: HardWorld
                y: 65.5
                x: -211.5
            _sign_HardWorld_nether_-13_33_35:
                icon: sign
                markup: true
                label: Portal to:<br/>PWiddy Cavern<br/>Not 100% safe
                z: 35.5
                world: HardWorld_nether
                y: 33.5
                x: -12.5
            _sign_HardWorld_-698_71_27:
                icon: sign
                markup: true
                label: Rose's<br/>Zombie Spawner
                z: 27.5
                world: HardWorld
                y: 71.5
                x: -697.5
            _sign_HardWorld_nether_-92_71_16:
                icon: sign
                markup: true
                label: Portal to:<br/>The Stronghold<br/>and back
                z: 16.5
                world: HardWorld_nether
                y: 71.5
                x: -91.5
            _sign_HardWorld_nether_250_61_-795:
                icon: sign
                markup: true
                label: Portal to:<br/>Northern<br/>Mushroom Island
                z: -794.5
                world: HardWorld_nether
                y: 61.5
                x: 250.5
            _sign_HardWorld_-4050_71_240:
                icon: sign
                markup: true
                label: WesterMost<br/>Gate
                z: 240.5
                world: HardWorld
                y: 71.5
                x: -4049.5
            _sign_HardWorld_nether_-76_70_53:
                icon: sign
                markup: true
                label: Portal to:<br/>Mt. Mactso
                z: 53.5
                world: HardWorld_nether
                y: 70.5
                x: -75.5
            _sign_HardWorld_nether_87_61_-615:
                icon: sign
                markup: true
                label: Portal to:<br/>Farpoint<br/>Village
                z: -614.5
                world: HardWorld_nether
                y: 61.5
                x: 87.5
            _sign_HardWorld_nether_-25_66_-5:
                icon: sign
                markup: true
                label: Portal to:<br/>Casa C7ron<br/>and back
                z: -4.5
                world: HardWorld_nether
                y: 66.5
                x: -24.5
            _sign_HardWorld_171_66_278:
                icon: sign
                markup: true
                label: Clay Trader's
                z: 278.5
                world: HardWorld
                y: 66.5
                x: 171.5
            _sign_HardWorld_193_65_1071:
                icon: sign
                markup: true
                label: Southern<br/>Horse Path
                z: 1071.5
                world: HardWorld
                y: 65.5
                x: 193.5
            _sign_HardWorld_nether_-89_72_4:
                icon: sign
                markup: true
                label: Portal to:<br/>Rose's Zombie<br/>Spawner &amp; back
                z: 4.5
                world: HardWorld_nether
                y: 72.5
                x: -88.5
            _sign_HardWorld_-1470_70_85:
                icon: sign
                markup: true
                label: Flaming Beach
                z: 85.5
                world: HardWorld
                y: 70.5
                x: -1469.5
            _sign_HardWorld_-729_20_152:
                icon: sign
                markup: true
                label: The Stronghold
                z: 152.5
                world: HardWorld
                y: 20.5
                x: -728.5
            _sign_HardWorld_8492_81_-148:
                icon: sign
                markup: true
                label: Test Sign
                z: -147.5
                world: HardWorld
                y: 81.5
                x: 8492.5
            _sign_HardWorld_-4204_68_69:
                icon: sign
                markup: true
                label: Ravine Above<br/>Ravine<br/>Cave
                z: 69.5
                world: HardWorld
                y: 68.5
                x: -4203.5
            _sign_HardWorld_1988_76_-887:
                icon: sign
                markup: true
                label: Mine<br/>Giant Cave
                z: -886.5
                world: HardWorld
                y: 76.5
                x: 1988.5
            _sign_HardWorld_nether_-97_76_40:
                icon: sign
                markup: true
                label: Portal to:<br/>Camp Hesiod
                z: 40.5
                world: HardWorld_nether
                y: 76.5
                x: -96.5
            _sign_HardWorld_297_65_313:
                icon: sign
                markup: true
                label: Jam's Place
                z: 313.5
                world: HardWorld
                y: 65.5
                x: 297.5
            _sign_HardWorld_950_64_9043:
                icon: sign
                markup: true
                label: Southern Ice<br/>Spikes<br/>Homestead
                z: 9043.5
                world: HardWorld
                y: 64.5
                x: 950.5
            _sign_HardWorld_nether_-9_65_32:
                icon: sign
                markup: true
                label: Portal to:<br/>PWiddy Castle
                z: 32.5
                world: HardWorld_nether
                y: 65.5
                x: -8.5
            _sign_HardWorld_-28_83_290:
                icon: sign
                markup: true
                label: Castle Widdy<br/>Powered by<br/>PWiddy!
                z: 290.5
                world: HardWorld
                y: 83.5
                x: -27.5
            _sign_HardWorld_-2908_64_-2499:
                icon: sign
                markup: true
                label: Hesipolis
                z: -2498.5
                world: HardWorld
                y: 64.5
                x: -2907.5
            _sign_HardWorld_129_77_301:
                icon: sign
                markup: true
                label: Wenda<br/>Weather God<br/>Temple
                z: 301.5
                world: HardWorld
                y: 77.5
                x: 129.5
            _sign_HardWorld_2086_65_-7417:
                icon: sign
                markup: true
                label: Cave!!<br/>North-NorthEast
                z: -7416.5
                world: HardWorld
                y: 65.5
                x: 2086.5
            _sign_HardWorld_1936_65_464:
                icon: sign
                markup: true
                label: River Source
                z: 464.5
                world: HardWorld
                y: 65.5
                x: 1936.5
            _sign_HardWorld_nether_-259_75_30:
                icon: sign
                markup: true
                label: Portal to:<br/>The Lost Wither<br/>Sailing Beach
                z: 30.5
                world: HardWorld_nether
                y: 75.5
                x: -258.5
            _sign_HardWorld_nether_-182_75_10:
                icon: sign
                markup: true
                label: Portal to:<br/>Flaming Beach
                z: 10.5
                world: HardWorld_nether
                y: 75.5
                x: -181.5
            _sign_HardWorld_-1936_70_-7402:
                icon: sign
                markup: true
                label: Sheer Drop<br/>to Vast<br/>Ice Cave
                z: -7401.5
                world: HardWorld
                y: 70.5
                x: -1935.5
            _sign_HardWorld_700_74_-2538:
                icon: sign
                markup: true
                label: Plymouth<br/>Powered by<br/>Belsean!
                z: -2537.5
                world: HardWorld
                y: 74.5
                x: 700.5
            _sign_HardWorld_-774_69_335:
                icon: sign
                markup: true
                label: Camp Hesiod<br/>&lt;-----
                z: 335.5
                world: HardWorld
                y: 69.5
                x: -773.5
            _sign_HardWorld_-1525_66_189:
                icon: sign
                markup: true
                label: WITHERDOME!
                z: 189.5
                world: HardWorld
                y: 66.5
                x: -1524.5
            _sign_HardWorld_-1505_68_1195:
                icon: sign
                markup: true
                label: SpruceWall<br/>Duckpond
                z: 1195.5
                world: HardWorld
                y: 68.5
                x: -1504.5
            _sign_HardWorld_nether_-303_75_31:
                icon: sign
                markup: true
                label: Fern Cave Gate
                z: 31.5
                world: HardWorld_nether
                y: 75.5
                x: -302.5
            _sign_HardWorld_nether_-69_67_25:
                icon: sign
                markup: true
                label: Portal to:<br/>Hill Gate from<br/>The Stronghold
                z: 25.5
                world: HardWorld_nether
                y: 67.5
                x: -68.5
            _sign_HardWorld_nether_-74_39_-1621:
                icon: sign
                markup: true
                label: Giant<br/>Nether Fortress
                z: -1620.5
                world: HardWorld_nether
                y: 39.5
                x: -73.5
            _sign_HardWorld_7446_66_-183:
                icon: sign
                markup: true
                label: Corkscrew Hill<br/>Village
                z: -182.5
                world: HardWorld
                y: 66.5
                x: 7446.5
            _sign_HardWorld_nether_-49_65_26:
                icon: sign
                markup: true
                label: Portal to:<br/>MactsoCo<br/>Trading Center
                z: 26.5
                world: HardWorld_nether
                y: 65.5
                x: -48.5
            _sign_HardWorld_4421_67_-118:
                icon: sign
                markup: true
                label: Eastern<br/>Ice<br/>Caves
                z: -117.5
                world: HardWorld
                y: 67.5
                x: 4421.5
        layerprio: 0
        areas: {
            }
        deficon: default
        lines: {
            }
        label: Markers
playersets: {
    }

**Comment by Psy-Virus** [comment]
Hey,
could you please take your config and marker file between `<pre></pre>` tags? Or something in that direction or post ist on [pastebin](Pastebin)?

**Comment by mactso** [comment]
Sure!
Transfering to pastbin now.

**Comment by mactso** [comment]
Wow!  Thanks.

There is a markers folder but it is empty on both the test server and the real server.
The marker.yml file is in minecraftserver\plugins\dynmap\markers.yml

There is no dynmap\tiles folder (so no dynmap/tiles/path/_markers_/  folder)

Sounds like I need to remove all traces of the old dynmap folder and let it install with the current version.

Because we were upgrading, perhaps something went wrong going from 1.9.1 to the new version.

It sounds like it will pick up the existing signs and rebuild the marker file?  Is that correct?

**Comment by Psy-Virus** [comment]
Well you misunderstood me.. sry, I think I wasn't clear enough. With the marker/tiles/path I meant that you should check, if the _markers_ folder in your tiles path exist; so in your case it should be dynmap/web/tiles/_markers_ ^^

And what I meant was that you should (after updating dynmap and spigot to latest versions) copy your markers.yml to any save place and let dynmap rebuild it after a server (re)start :)
then you should check, if you can create markers. If so, then your markers.yml has anything in it what dynmap doesn't like^^ Also if I can't find anything except that it isn't well formatted, but that's not a problem.

Btw: On my machine it wasn't any problem to update dynmap. I never had any problems with it. except [that one](https://github.com/webbukkit/dynmap/issues/1481), @mikeprimm won't investigate (yes mike, if you see this, it's now over a year old ;) )

And you also should clean your browsers cache ->[How to](http://www.wikihow.com/Clear-Your-Browser%27s-Cache). Only ctrl+F5 isn't ever good enough ;)

**Comment by Psy-Virus** [comment]
Oh and it also could lay on any other mod you've installed. So you also should check if there's something interfering with dynmap by disabling all mods and enable them one after another.

**Comment by mactso** [comment]
Summary: clearing the web cache fixed it.  Thanks a BUNCH.

details
Okay, yes the web/tiles/markers exists and has 88 files.
This is the same as in my test server (where the signs work).
I set the markers.yml aside to a save folder and created a sign.
Didn't work.
Cleared the cache from your instructions and that was it.
Tested it quickly by loading IE (which I never use) and it had signs so then I cleared firefox.
I've passed the tip on to our other players.

I'll be using your updated config file tomorrow when I have more time.  Thanks again for that.



Thanks!  I will be working on this tomorrow night again.
